doc.go

```diff
@@ -797,28 +797,17 @@ and before retrieving the results.
 		...
 
 		func DemonstrateAsyncMode(db *sql.DB) {
-			// Enable asynchronous mode.
-			ctx := WithAsyncMode(context.Background())
-			// Establish connection
-			conn, _ := db.Conn(ctx)
-
-			// Unwrap connection
-			err = conn.Raw(func(x interface{}) error {
-				// Execute asynchronous query
-				rows, _ := x.(driver.QueryerContext).QueryContext(ctx, "select 1", nil)
-				defer rows.Close()
-
-				// Retrieve and check results of the query after casting the result
-				status := rows.(SnowflakeResult).GetStatus()
-				if status == QueryStatusComplete {
-					// do something
-				} else if status == QueryStatusInProgress {
-					// do something
-				} else if status == QueryFailed {
-					// do something
-				}
-				return nil
-			})
+			// Enable asynchronous mode
+			ctx := sf.WithAsyncMode(context.Background())
+
+			// Run the query with asynchronous context
+			rows, _ := db.QueryContext(ctx, "select 1")
+
+			// do something as the workflow continues whereas the query is computing in the background
+			...
+
+			// Get the data when you are ready to handle it
+			cols, err := rows.Columns()
```

Comment:
Why fetching columns?

Reply:
I took it from the `cmd/async.go` example. Replaced it with `.Scan()`.